eventuate-tram-messaging-proxy-service/src/main/java/io/eventuate/tram/messaging/proxy/service/SubscriptionController.java

```diff
@@ -9,37 +9,65 @@
 import org.springframework.web.client.RestTemplate;
 
 import java.util.Optional;
+import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 
 @RestController
 public class SubscriptionController {
 
+  @Autowired
+  private ProxyProperties proxyProperties;
+
   @Autowired
   private MessageConsumerImplementation messageConsumerImplementation;
 
   @Autowired
   private RestTemplate restTemplate;
 
   private ConcurrentMap<String, MessageSubscription> messageSubscriptions = new ConcurrentHashMap<>();
+  private ConcurrentMap<String, Long> messageSubscriptionUpdateTime = new ConcurrentHashMap<>();
```

Comment:
Two maps with the same key suggest that you should have a single map containing some new class.

Reply:
I will fix